hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1442,7 +1477,31 @@ protected URI canonicalizeUri(URI rawUri) {
   @Retries.RetryTranslated
   public FSDataInputStream open(Path f, int bufferSize)
       throws IOException {
-    return open(f, Optional.empty(), Optional.empty());
+    if (this.prefetchEnabled) {
+      return this.createPrefetchingInputStream(f, bufferSize);
+    } else {
+      return open(f, Optional.empty(), Optional.empty());
```

Comment:
This must go in the open/3 method below, so that openFile() will also use it.
You will also get the benefits of the filestatus retrieved during the existence checks, with all length and etag; openFile() allows callers to pass it in so we can skip the HEAD request.